ggshield/scan/scannable.py

```diff
@@ -178,29 +178,37 @@ def get_all_results(self) -> Iterable[Result]:
 class File:
     """Class representing a simple file."""
 
-    def __init__(self, document: str, filename: str):
-        self.document = document
+    def __init__(self, document: Optional[str], filename: str):
+        self._document = document
         self.filename = filename
         self.filemode = Filemode.FILE
 
     def relative_to(self, root_path: Path) -> "File":
-        return File(self.document, str(Path(self.filename).relative_to(root_path)))
+        return File(self._document, str(Path(self.filename).relative_to(root_path)))
+
+    @property
+    def document(self) -> str:
+        self.prepare()
+        assert self._document is not None
```

Comment:
Is this necessary ? We will always have the result of a `File._decode_bytes` which should be a str ?

Reply:
This is a way to make mypy aware that we know `self._document` is an str now. Without the assert it reports that `self._document` may be None. I prefer this to a `#  type: ignore` comment because it documents assumptions.